pkg/stripe/analytics_telemetry.go

```diff
@@ -0,0 +1,169 @@
+package stripe
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/google/uuid"
+
+	"github.com/spf13/cobra"
+
+	"github.com/stripe/stripe-cli/pkg/useragent"
+	"github.com/stripe/stripe-cli/pkg/version"
+)
+
+//
+// Public types
+//
+
+// TelemetryContextKey is the key for the telemetry context
+type TelemetryContextKey struct{}
+
+// TelemetryClientKey is the key for the telemetry client
+type TelemetryClientKey struct{}
+
+// DefaultTelemetryEndpoint is the default URL for the telemetry destination
+const DefaultTelemetryEndpoint = "https://r.stripe.com/0"
+
+// CLIAnalyticsEventContext is the structure that holds telemetry data context that is ultimately sent to the Stripe Analytics Service.
+type CLIAnalyticsEventContext struct {
+	InvocationID      string `url:"invocation_id"`      // The invocation id is unique to each context object and represents all events coming from one command / gRPC method call
+	UserAgent         string `url:"user_agent"`         // the application that is used to create this request
+	CommandPath       string `url:"command_path"`       // the command or gRPC method that initiated this request
+	Merchant          string `url:"merchant"`           // the merchant ID: ex. acct_xxxx
+	CLIVersion        string `url:"cli_version"`        // the version of the CLI
+	OS                string `url:"os"`                 // the OS of the system
+	GeneratedResource bool   `url:"generated_resource"` // whether or not this was a generated resource
+}
+
+// TelemetryClient is an interface that can send two types of events: an API request, and just general events.
+type TelemetryClient interface {
+	SendAPIRequestEvent(ctx context.Context, requestID string, livemode bool) (*http.Response, error)
+	SendEvent(ctx context.Context, eventName string, eventValue string) (*http.Response, error)
+}
+
+// AnalyticsTelemetryClient sends event information to r.stripe.com
+type AnalyticsTelemetryClient struct {
+	BaseURL    *url.URL
+	WG         *sync.WaitGroup
+	HTTPClient *http.Client
+}
+
+//
+// Public functions
+//
+
+// InitContext initializes an instance of CLIAnalyticsEventContext
+func InitContext() *CLIAnalyticsEventContext {
```

Comment:
`NewContext` would be a more natural name in Go for this function.

Reply:
updated